clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -111,7 +112,7 @@ def CIR_StructType : CIR_Type<"Struct", "struct",
     "bool":$body,
     "bool":$packed,
     "mlir::cir::StructType::RecordKind":$kind,
-    "std::optional<::mlir::cir::ASTRecordDeclAttr>":$ast
+    OptionalParameter<"std::optional<ASTRecordDeclInterface>">:$ast
```

Comment:
Now that this becomes a OptionalParameter, do we really need to keep the `std::optional` inside?